pdfwutils.py

```diff
@@ -1078,7 +1078,29 @@ def add_multi_imagepage(
                 image1[PdfName.Height] = -Im_i['imgheightpx']
             else:
                 image1[PdfName.Height] = Im_i['imgheightpx']
-            image1[PdfName.ColorSpace] = PdfName.DeviceRGB
+
+            if Im_i['color'] == Colorspace["1"] or Im_i['color'] == Colorspace.L:
+                image1[PdfName.ColorSpace] = PdfName.DeviceGray
+            elif Im_i['color'] == Colorspace.RGB:
+                image1[PdfName.ColorSpace] = PdfName.DeviceRGB
+            elif Im_i['color'] == Colorspace.CMYK or Im_i['color'] == Colorspace["CMYK;I"]:
+                image1[PdfName.ColorSpace] = PdfName.DeviceCMYK
```

Comment:
There are no 1-bit jpegs , and cmyk jpegs are extremely rare, so this can be a lot simpler.